src/components/data-table/PDataTable.vue

```diff
@@ -0,0 +1,71 @@
+<template>
+  <table :class="classNames">
+    <p-data-table-row v-if="!noHeaders">
+      <p-data-table-header
+        v-for="(header, k) in headers"
+        :key="`p-td-${k}`"
+        @click="$emit('header-click', { header })">
```

Comment:
@iagocavalcante `header-click` or `headerClick`?

Reply:
header-click in this case!